packages/babel-helpers/src/helpers/applyDecs2311.ts

```diff
@@ -334,100 +334,98 @@ export default /* @no-mangle */ function applyDecs2311(
         }.bind(null, decoratorFinishedRef),
       };
 
-      try {
```

Comment:
It _might_ be observable in this test case (I'm not sure):
```js
let addInitializer;
function foo(ctx) {
  addInitializer = ctx.addInitializer;
  throw {};
}
try {
  class A { @dec foo = 2; }
} catch {}

expect(() => addInitializer(() => {})).toThrow();
```

Reply:
Yes, this can be observed.
But when the decorator throws an error, it seems that this check is no longer important.
What do you think?

Reply:
Well, we should make sure to match the spec :)

https://ci.tc39.es/preview/tc39/ecma262/sha/59003ed603d09485b957a0f579b46697f49a96a6/#sec-applydecoratorstoelementdefinition

It looks like if calling the decorator throws then `decoratorFinishedRef` is not set to `true`, because the ? in step of 5.j is an early return in case of error. So removing the try/catch is indeed correct.

However, we should move the `decoratorFinishedRef.v` assignment to right after the `newValue = dec.call(` call, so that if one of the following validation steps throws then we still have it set to `true`.

Then, please add a test for when the decorator throws (`addInitializer` should still be callable) and for when one of the subsequent checks throws (`addInitializer` should not be callable anymore) :)